velox/functions/lib/tests/TimeUtilsTest.cpp

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/lib/TimeUtils.h"
+#include <gtest/gtest.h>
+
+namespace facebook::velox::functions {
+
+class FirstDayOfWeekTest
+    : public testing::TestWithParam<std::tuple<int32_t, uint32_t, uint32_t>> {};
+
+TEST_P(FirstDayOfWeekTest, getFirstDayOfWeek) {
+  auto& [y, firstDayOfWeek, res] = GetParam();
+  ASSERT_EQ(getDayOfFirstDayOfWeek(y, firstDayOfWeek), res);
+}
+
+INSTANTIATE_TEST_CASE_P(
+    WeekYear,
+    FirstDayOfWeekTest,
+    testing::Values(
+        std::make_tuple(2024, 1, 7),
+        std::make_tuple(2024, 2, 1),
+        std::make_tuple(2024, 3, 2),
+        std::make_tuple(2024, 4, 3),
+        std::make_tuple(2024, 5, 4),
+        std::make_tuple(2024, 6, 5),
+        std::make_tuple(2024, 7, 6)));
```

Comment:
I wonder if it is easy to locate the buggy case. Could you share the log when there is some test failing? The common way is to use ASSERT_EQ to test each case.